Mathlib/CategoryTheory/Abelian/InjectiveResolution.lean

```diff
@@ -254,45 +254,67 @@ namespace InjectiveResolution
 
 /-!
 Our goal is to define `InjectiveResolution.of Z : InjectiveResolution Z`.
-The `0`-th object in this resolution will just be `injective.under Z`,
+The `0`-th object in this resolution will just be `Injective.under Z`,
 i.e. an arbitrarily chosen injective object with a map from `Z`.
-After that, we build the `n+1`-st object as `injective.syzygies`
+After that, we build the `n+1`-st object as `Injective.syzygies`
 applied to the previously constructed morphism,
-and the map from the `n`-th object as `injective.d`.
+and the map from the `n`-th object as `Injective.d`.
 -/
 
 
 variable [Abelian C] [EnoughInjectives C]
 
 /-- Auxiliary definition for `InjectiveResolution.of`. -/
-@[simps]
+@[simps!]
 def ofCocomplex (Z : C) : CochainComplex C ℕ :=
   CochainComplex.mk' (Injective.under Z) (Injective.syzygies (Injective.ι Z))
-    (Injective.d (Injective.ι Z)) fun ⟨X, Y, f⟩ =>
+    (Injective.d (Injective.ι Z)) fun ⟨_, _, f⟩ =>
     ⟨Injective.syzygies f, Injective.d f, (exact_f_d f).w⟩
+set_option linter.uppercaseLean3 false in
 #align category_theory.InjectiveResolution.of_cocomplex CategoryTheory.InjectiveResolution.ofCocomplex
 
+-- Porting note: removed this from `of` due to timeouts
+theorem ofCocomplex_sq_01_comm (Z : C) :
+    Injective.ι Z ≫ HomologicalComplex.d (ofCocomplex Z) 0 1 =
+    HomologicalComplex.d ((CochainComplex.single₀ C).obj Z) 0 1 ≫ 0 := by
+  simp only [ofCocomplex_d, eq_self_iff_true, eqToHom_refl, Category.comp_id,
+    dite_eq_ite, if_true, comp_zero]
+  exact (exact_f_d (Injective.ι Z)).w
+
+-- Porting note: removed this from `of` due to timeouts
+theorem exact_ofCocomplex (Z : C) (n : ℕ) :
+    Exact (HomologicalComplex.d (ofCocomplex Z) n (n + 1))
+    (HomologicalComplex.d (ofCocomplex Z) (n + 1) (n + 2)) :=
+  match n with
+-- Porting note: used to be simp; apply exact_f_d on both branches
+    | 0 => by simp; apply exact_f_d
+    | m+1 => by
+      simp only [ofCocomplex_X, ComplexShape.up_Rel, not_true, ofCocomplex_d,
+        eqToHom_refl, Category.comp_id, dite_eq_ite, ite_true]
+      erw [if_pos (c := m + 1 + 1 + 1 = m + 2 + 1) rfl]
+      apply exact_f_d
```

Comment:
I don't understand the porting notes. If I look at https://github.com/leanprover-community/mathlib4/pull/4059/files/55e7d1934b1a85ac70cf2a3e6e8c0d2060974b0b..c43ab2d40a78d0a88c88511c610f980b8208a3ab I don't see what this code corresponds to in the mathport output. Could you please expand a little?

Reply:
I updated the porting notes to be more informative. TLDR: I outlined from the `of` declaration these two results because `of` was incredibly slow.